File: java/007. Reverse Integer.java
// 7. Reverse Integer
// Accepted 28ms

class Solution {
    public int reverse(int x) {
        long rx = 0L;
        boolean isNegative = (x < 0);
        
        if(isNegative)
            x *= -1;
        
        while(x > 0) {
            rx = (rx * 10L) + (x % 10);
            x /= 10;
        }
        
        if(isNegative)
            rx *= -1L;
        
        if(rx < Integer.MIN_VALUE || rx > Integer.MAX_VALUE)
            rx = 0L;
        
        return (int)rx;
    }
}

File: python3/007. Reverse Integer.py
# 7. Reverse Integer
# Accepted 52ms

class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x == 0: return 0
        
        if x < 0:
            x = -x; isNeg = True
        else:
            isNeg = False
        rev = 0
        
        while x > 0:
            rev = (rev * 10) + (x % 10)
            x //= 10
        
        if isNeg: rev = -rev
        if rev > 2147483647 or rev < -2147483648: rev = 0
        
        return rev
